src/ui/hooks/useGetCollaboratorNames.ts

```diff
@@ -0,0 +1,114 @@
+import { useContext, useEffect, useMemo, useState } from "react";
+
+import { SessionContext } from "bvaughn-architecture-demo/src/contexts/SessionContext";
+import { GetWorkspaceMembers, GetWorkspaceMembersVariables } from "graphql/GetWorkspaceMembers";
+import {
+  useGetOwnersAndCollaborators,
+  useGetRecording,
+  useGetRecordingId,
+} from "ui/hooks/recordings";
+import { GET_WORKSPACE_MEMBERS } from "ui/hooks/workspaces_user";
+import { query } from "ui/utils/apolloClient";
+
+export type PartialUser = {
+  id: string;
+  name: string;
+};
+export type PartialUsers = PartialUser[];
+
+const EMPTY_ARRAY: PartialUsers = [];
+
+// Returns a unified list of collaborators and workspace members with access to the current recording.
+export default function useRecordingUsers(excludeCurrentUser: boolean): PartialUser[] {
+  const { currentUserInfo } = useContext(SessionContext);
+
+  // This info is loaded by a nested component because we can't conditionally load it otherwise.
+  const [workspaceUsers, setWorkspaceUsers] = useState<PartialUsers>(EMPTY_ARRAY);
```

Comment:
This would not be necessary if you use a query that starts with the `recordingId` instead of the `workspaceId`. For example you could fetch all the collaborator names with a single query like this:
```graphql
query GetCollaboratorNames($recordingId: UUID!) {
  recording(uuid: $recordingId) {
    owner {
      id
      name
    }
    collaborators {
      edges {
        node {
          ... on RecordingUserCollaborator {
            user {
              id
              name
            }
          }
        }
      }
    }
    workspace {
      members {
        edges {
          node {
            ... on WorkspaceUserMember {
              user {
                id
                name
              }
            }
          }
        }
      }
    }
  }
}
```
(I left out the `RecordingPendingUserCollaborator`s here even though they are included by your PR - do we want to show them or leave them out like the `WorkspacePendingUserMember`s?)

Reply:
> (I left out the `RecordingPendingUserCollaborator`s here even though they are included by your PR - do we want to show them or leave them out like the `WorkspacePendingUserMember`s?)

The main reason I left them out was that they didn't have _names_ (so nothing to filter on).

Reply:
`WorkspacePendingUserMember` does have a `name` (although it's optional, which I think it shouldn't be), only `WorkspacePendingEmailMember` doesn't have a name.
But I guess we should leave out all pending collaborators and members since they haven't agreed yet to be a collaborator/member.